### PR TITLE
Initial docstrings and doctests for inference methods

### PR DESCRIPTION
**Pull request by v-a-s-a**
This was trickier to setup than I thought it would be, but it seemed worth submitting for comment. There are different ways to setup the doctest that may be cleaner. Thoughts and comments:
- I turned off the progress printing and verbosity. This allowed me to control the output of the doctest. These should be passed in as explicit keyword arguments, or perhaps using `kwargs`. A potential alternative is to user the doctest output filters, but I wasn't able to get these working for this example.
- I also switched travis to build docs without printing deprecation warnings. There were a few of these in thrown deep within `advancedHMC`  when using defaults and causing the test to fail. It may be possible to just construct a call to sample that uses none of the deprecated arguments.

### CODE REVIEW

**Comment by v-a-s-a** [comment]
I've reverted back to flagging deprecation warnings. I could build docs with a custom flag, but apparently running the doctests is different. Instead, I pulled the updated interface from the `advancedHMC` README.

**Comment by cscherrer** [comment]
Great, thank you! The current AdvancedHMC interface definitely needed some updating, exciting to see this moving ahead

**Comment by v-a-s-a** [comment]
I couldn't quite understand what `method = logpdf` was doing in `dynamicHMC`. Will have to circle back when my understanding of Soss primitives improves.

**Comment by cscherrer** [comment]
> I couldn't quite understand what `method = logpdf` was doing in `dynamicHMC`. 

It's not you, this one is a bit tricky.

DynamicHMC requires computation of the `logpdf`. By default, Soss computes the `logpdf` of a model by adding the `logpdf`s of the components. But in some cases, this can be done more efficiently with some symbolic simplifications. So we can use a SymPy representation and do some rewrites (mostly these are in Julia), and then generate code for this. So the current options here are `method=logpdf` or `method=codegen`.

The SymPy stuff is currently very limited, and I'd like to move toward using `SymbolicUtils.jl` as that matures.

File: src/inference/dynamicHMC.jl [hunk 1]
```diff
@@ -9,6 +9,69 @@ import LogDensityProblems: ADgradient
...
+
+## Arguments
+* `rng`: Random number generator.
+* `m`: Soss model.
+* `_data`: `NamedTuple` of data to condition on.
+
+
+## Keyword Arguments
+*  `N = 1000`: Number of samples to draw.
+*  `method = logpdf`: .
```
Comment (cscherrer) [inline src/inference/dynamicHMC.jl:36]:
```suggestion
*  `method = logpdf`: How to compute the log-density. Options are `logpdf` (delegates to `logpdf` of each component) or `codegen` (symbolic simplification and code generation).
```

**Pull request merged**